source/paddle.js

```diff
@@ -889,6 +1006,503 @@ paddle.AttributeType = {
+        }
+        return denotation.join(';');
+    }
+};
+
+paddle.IR.OpInfoData = class extends paddle.IR.OpInfo {
+
+    init(op) {
+        const [opKey, opType] = this._name.split('.');
+        this._opKey = opKey;
+        this._opType = opType;
+        let prefix = '';
+        for (const attr of op.A) {
+            if (attr.N === 'name') {
+                prefix = attr.AT.D;
+                break;
+            }
+        }
+        this._prefix = prefix;
+        this._attr = op.A;
+    }
+
+    get namePrefix() {
+        return this._prefix;
+    }
+
+    getOutputAttr(idx, outputAttr) {
+        const denotation = [];
+        for (const attr of outputAttr) {
+            const attrName = attr.N;
+            const attrValue = attr.AT.D[idx].D;
+            denotation.push(`${attrName}:${attrValue}`);
+        }
+
+        for (const value of this._attr) {
+            const [attrName, , attrValue] = this.getAttr(null, value);
+            denotation.push(`${attrName}:${attrValue}`);
+        }
+
+        return denotation.join(';');
+    }
+};
+
+paddle.IR.OpInfoP = class extends paddle.IR.OpInfo {
+
+    init(op) {
+        const [name] = op.A.slice(3);
+        this._name = name;
+        this._type = paddle.Utility.getIRCompressOp(this._type);
+        this._fullName = this._type;
+        op.OA = [...op.OA, ...op.A];
+    }
+
+    get type() {
+        return this._type;
+    }
+
+    get name() {
+        return this._type;
+    }
+
+    get namePrefix() {
+        return this._name;
+    }
+
+    get fullName() {
+        return this._fullName;
+    }
+
+    getAttr(idx, value) {
+        let attrName = '';
+        let attrType = '';
+        let attrValue = '';
+        switch (idx) {
+            case '0':
+                attrName = 'is_distributed';
+                attrType = paddle.Utility.getIRType('a_bool');
+                break;
+            case '1':
+                attrName = 'is_parameter';
+                attrType = paddle.Utility.getIRType('a_bool');
+                break;
+            case '2':
+                attrName = 'need_clip';
+                attrType = paddle.Utility.getIRType('a_bool');
+                break;
+            case '3':
+                attrName = 'name';
+                attrType = paddle.Utility.getIRType('a_str');
+                break;
+            default:
+                break;
+        }
+        attrValue = attrType === paddle.Utility.getIRType('a_bool') ? value === 1 : value;
+        return [attrName, attrType, attrValue];
+    }
+
+    getOutputAttr(idx, outputAttr) {
+        const persistable = outputAttr[0] === 1;
+        const stop_gradient = outputAttr[1] === 1;
+        const trainable = outputAttr[2] === 1;
+        const is_distributed = outputAttr[3] === 1;
+        const is_parameter = outputAttr[4] === 1;
+        const need_clip = outputAttr[5] === 1;
+        const denotation = [
+            `persistable:${persistable}`,
```

Comment:
These strings might become rather long. Would it work to include a term like `persistable` if `true` and omit it if `false` to have a shorter string of used flags.

Reply:
Done! Only show `true` attrs ~ Thanks!